plugins/rest/auth.go

```diff
@@ -629,8 +661,30 @@ func (ap *awsSigningAuthPlugin) validateConfig() error {
 		}
 	}
 
-	if ap.AWSService == "" {
-		ap.AWSService = awsSigv4SigningDefaultService
+	ap.AWSService = strings.ToLower(ap.AWSService)
+
+	// Only allow ECR for OCI service types
+	if serviceType == "oci" {
+		if ap.AWSService == "" {
+			ap.AWSService = "ecr"
+		}
+
+		if ap.AWSService != "ecr" {
+			return fmt.Errorf("cannot use aws service %q with ecr", ap.AWSService)
+		}
+
+		// We need to setup a special auth plugin for ECR.
+		ap.ecrAuthPlugin = newECRAuthPlugin(ap)
+	}
```

Comment:
Nit: An `else` block should work here.

Reply:
This was intentional. I think the conditional bodies were just so big that I found it helpful to reiterate the condition. But this is just my opinion on style of course.

Addressed in https://github.com/open-policy-agent/opa/pull/5560/commits/c12cb66519f0d27c7addfea20bcfdd91f6c67ff3